File: 24-QUE/W-Find Greatest common Divisor of array/Gcd.java

import java.util.*;

public class Gcd{

public static void Gcd(int s , int l){
        int  gcd = 0;
        for(int i=1; i<=s && i<=l; i++){
            if(s%i ==0 && l%i ==0 ){
                gcd = i;
            }
        } 
        System.out.println("GCD of  "+ s +"and  " + l + "is  " +gcd);
}
public static void main(String args[]){

    Scanner sc = new Scanner(System.in); 
    System.out.println("enter the size of array");
    int n = sc.nextInt(); 

    int arr[] = new int[n]; 

    for(int i =0;i<n;i++){
        arr[i] = sc.nextInt();  
     } 
    Arrays.sort(arr); 
    System.out.println("Smallest number" + arr[0]); 
    System.out.println("largest number" + arr[n-1]);  
    int  s = arr[0];
    int  l = arr[n-1];

    Gcd(s,l);

}

}

File: 24-QUE/W-Find Greatest common Divisor of array/Gcd.c

#include<stdio.h>
#include<conio.h> 

int main(){
    int n,temp,i,j,turn,k, gcd;  
  int arr[20]; 
  int s , l;
    
  printf("enter the size of array"); 
  scanf("%d",&n);  

  for( i=0; i<n;i++){
    printf("enter the elements");
    scanf("%d", &arr[i]);
  } 

    //sort 

    for(turn=0; turn<n ;turn ++)
    for(j = 0 ; j<n ;j++){
        if(arr[j]> arr[j+1]){
            temp = arr[j]; 
            arr[j] = arr[j+1]; 
            arr[j+1] = temp;
        }
    } 


    printf("sorted array");
    for( i=0;i<n-1;i++){
        printf("%d\t",arr[i]);
    }  

    s = arr[0]; 
    l = arr[n-1];  

    printf("smallest number %d\n  largest number %d", s,l);
     
    // Gcd 
    for(k=1; k<=s && k<=l ;k++){
        if(s%k ==0 && l%k ==0){
            gcd = k;
        }
    } 
    printf("GCD is %d\n ",gcd);

    return 0 ;
}